src/main/java/process/publication/ScheduleFileExport.java

```diff
@@ -0,0 +1,108 @@
+package process.publication;
+
+import models.Model;
+import models.Schedule;
+import models.Server;
+import models.Session;
+import net.fortuna.ical4j.model.DateTime;
+import net.fortuna.ical4j.model.TimeZone;
+import net.fortuna.ical4j.model.TimeZoneRegistryFactory;
+import net.fortuna.ical4j.model.component.VEvent;
+import net.fortuna.ical4j.model.property.Description;
+import net.fortuna.ical4j.model.property.Location;
+import net.fortuna.ical4j.model.property.ProdId;
+import net.fortuna.ical4j.model.property.TzId;
+import org.apache.logging.log4j.Logger;
+import utils.DateUtils;
+import utils.LoggerUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+
+/**
+ * Process permettant l'export et la publication de l'emploi du temps sous forme d'un fichier iCal.
+ */
+public class ScheduleFileExport extends Publication {
+
+  private static final Logger LOGGER = LoggerUtils.buildLogger(ScheduleFileExport.class);
+
+  /**
+   * Publie sur un serveur et un channel donné l'emploi du temps sous la forme d'un fichier iCal (.ics).
+   *
+   * @param serverRef   Réf. du serveur sur lequel faire la publication.
+   * @param channelName Nom du channel sur lequel le fichier obtenu sera publié.
+   * @return {@code true} si la publication s'est faite avec succès, {@code false} si un problème a empêché la publication.
+   */
+  public boolean export(String serverRef, String channelName) {
+    // Utilise la date actuelle
+    return this.export(serverRef, channelName, Calendar.getInstance().getTime());
+  }
+
+  /**
+   * Publie sur un serveur et un channel donné l'emploi du temps à partir de la date donnée sous la forme d'un fichier iCal (.ics).
+   *
+   * @param serverRef   Réf. du serveur sur lequel faire la publication.
+   * @param channelName Nom du channel sur lequel le fichier obtenu sera publié.
+   * @param fromDate    Date à laquelle l'emploi du temps publiée commencera.
+   * @return {@code true} si la publication s'est faite avec succès, {@code false} si un problème a empêché la publication.
+   */
+  public boolean export(String serverRef, String channelName, Date fromDate) {
```

Comment:
Peut être extraire les opérations commentées dans des sous méthodes privées ?

Reply:
Ah je viens de voir que c'était en "work in progress", je n'ai pas fait gaffe comme j'ai eut la notif pour faire la review !